### PR TITLE
Fix nightly test for EXO

### PR DESCRIPTION
**Pull request by ganglyu**
### Description of PR

Summary:
Fix traffic issue on fanout switch.
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

#### What is the motivation for this PR?
Nightly test for EXO is broken.
Microsoft ADO: 25579175

#### How did you do it?
1. Do not use duthost directly
2. Fix mac conflict on fanout switch

#### How did you verify/test it?
Run end to end test

### CODE REVIEW

**Comment by bingwang-ms** [comment]
Please add `PR summary` and `Type of change`.
Why do we need to enable this test on dualtor?

**Comment by ganglyu** [comment]
> Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?

I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.

**Comment by bingwang-ms** [comment]
> > Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?
> 
> I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.

If we don't need to enable this test on dualtor, why do we need this change?
```
Support dualtor
```

**Comment by ganglyu** [comment]
> > > Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?
> > 
> > 
> > I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.
> 
> If we don't need to enable this test on dualtor, why do we need this change?
> 
> ```
> Support dualtor
> ```

It's a bug fix, we should not use duthost with rand_selected_dut

File: tests/acl/test_acl_outer_vlan.py [hunk 1]
```diff
@@ -557,7 +529,11 @@ def _do_verification(self, ptfadapter, duthost, tbinfo, vlan_setup_info, ip_vers
 
         pkt_type = QINQ if stage == INGRESS else None
         src_mac = ptfadapter.dataplane.get_mac(0, src_port[0])
-        dst_mac = test_setup_config.get('dst_mac', ptfadapter.dataplane.get_mac(0, dst_port[0]))
+        if stage == INGRESS:
+            # Use broadcast for ingress test
+            dst_mac = "ff:ff:ff:ff:ff:ff"
```
Comment (bingwang-ms) [inline tests/acl/test_acl_outer_vlan.py:534]: Can you please clarify why broadcast MAC is used for ingress test?

**Comment by bingwang-ms** [comment]
> > > > Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?
> > > 
> > > 
> > > I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.
> > 
> > 
> > If we don't need to enable this test on dualtor, why do we need this change?
> > ```
> > Support dualtor
> > ```
> 
> It's a bug fix, we should not use duthost with rand_selected_dut, and I have updated the description.

Discussed offline, let's keep using rand_selected_dut. It's the same with duthost on single tor testbed.

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**